docs/csharp/local-functions-vs-lambdas.md

```diff
@@ -53,6 +49,21 @@ the lambda expression, `nthFactorial` before defining it. Not doing so results
 in a compile time error for referencing `nthFactorial` before assigning it.
 Recursive algorithms are easier to create using local functions.
 
+Third, for lambda expressions, the compiler must always create an anonymous class
+and an instance of that class to store any variables captured by the
+closure. Consider this async example:
+
+[!code-csharp[TaskLambdaExample](../../samples/snippets/csharp/new-in-7/AsyncWork.cs#36_TaskLambdaExample "Task returning method with lambda expression")]
+
+The closure for this lambda expression contains the `address`,
+`index` and `name` variables. In the case of local functions, the object
+that implements the closure may be a `struct` type. That would save on
+an allocation. (The local function equivalent of this method also
+uses a class for the closure. It is an implementation detial which will
+be used for local functions):
```

Comment:
The punctuation here is weird: The second sentence in the parentheses does not end with a period and I think the colon does not really fit here.